### PR TITLE
Sambamba markdup

### PR DESCRIPTION
**Pull request by BioInf2305**
## PR checklist

Closes #XXX 
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/sambamba/markdup/main.nf [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda "bioconda::sambamba=1.0"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity//sambamba:1.0--h98b6b92_0':
+        'quay.io/biocontainers/sambamba:1.0--h98b6b92_0' }"
+
+    input:
+        tuple val(meta), path(bam)
```
Comment (nvnieuwk) [inline modules/nf-core/sambamba/markdup/main.nf:11]:
```suggestion
    tuple val(meta), path(bam)
```

To stay consistent with the other modules

File: modules/nf-core/sambamba/markdup/main.nf [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    output:
+    tuple val(meta), path("*.mrkDup.bam"), emit: bam
```
Comment (nvnieuwk) [inline modules/nf-core/sambamba/markdup/main.nf:14]:
```suggestion
    tuple val(meta), path("*.bam"), emit: bam
```

It's preferred to not add custom suffixes to the filename. It's better to let the user do this via the `task.ext.prefix` variable. You can always add a check to see if the input filename is the same as the output filename (and let it throw an error if it is the same) => paragraph 6 here: https://nf-co.re/docs/contributing/modules#naming-conventions

**Review by nvnieuwk — CHANGES_REQUESTED** [review]
Some small suggestions

File: modules/nf-core/sambamba/markdup/main.nf [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
+    """
+    sambamba \\
+        markdup \\
+        $args \\
+        -t $task.cpus \\
+        $bam \\
+        ${prefix}.mrkDup.bam
```
Comment (matthdsm) [inline modules/nf-core/sambamba/markdup/main.nf:30]:
```suggestion
        ${prefix}.bam
```

File: modules/nf-core/sambamba/markdup/meta.yml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        Groovy Map containing sample information
+        e.g. [ id:'test', single_end:false ]
+  - versions:
+      type: file
+      description: File containing software versions
+      pattern: "versions.yml"
+  - bam:
+      type: file
+      description: Sorted BAM/CRAM/SAM file
+      pattern: "*.mrkDup.{bam,cram,sam}"
```
Comment (matthdsm) [inline modules/nf-core/sambamba/markdup/meta.yml:37]:
```suggestion
      pattern: "*.{bam,cram,sam}"
```

**Review by nvnieuwk — APPROVED** [review]
LGTM, I would also suggest to maybe add a check that looks if the input and output file have the same name (see [here](https://nf-co.re/docs/contributing/modules#naming-conventions) part 6)

**Pull request merged**